src/VisualStudio/IntegrationTest/New.IntegrationTests/VisualBasic/BasicErrorListCommon.cs

```diff
@@ -128,13 +128,20 @@ public virtual async Task BuildErrorsInClosedFiles()
 
             // Verify the build error is listed in the error list for closed file.
             await TestServices.ErrorList.ShowBuildErrorsAsync(HangMitigatingCancellationToken);
-            var actualErrors = await TestServices.ErrorList.GetBuildErrorsAsync(HangMitigatingCancellationToken);
             var expectedErrors = new[] {
                 "Class1.vb(1, 1): error BC30481: 'Class' statement must end with a matching 'End Class'.",
             };
-            AssertEx.EqualOrDiff(
-                string.Join(Environment.NewLine, expectedErrors),
-                string.Join(Environment.NewLine, actualErrors));
+
+            while (true)
+            {
+                this.HangMitigatingCancellationToken.ThrowIfCancellationRequested();
+
+                var actualErrors = await TestServices.ErrorList.GetBuildErrorsAsync(HangMitigatingCancellationToken);
+                if (string.Join(Environment.NewLine, expectedErrors) == string.Join(Environment.NewLine, actualErrors))
+                    return;
+
+                await Task.Delay(TimeSpan.FromSeconds(1));
+            }
```

Comment:
no current way to ask lsp when it's done updating the UI.  so polling for now.

Reply:
We have code in Razor that uses events and a semaphore to wait for errors in a _slightly_ better way, if you're interested in future. I suspect neither is ideal though :)

https://github.com/dotnet/razor/blob/main/src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/InProcess/ErrorListInProcess.cs#L21